python/lsst/obs/base/ingest.py

```diff
@@ -355,16 +419,202 @@ def _calculate_dataset_info(self, header, filename):
+        """Given a list of files, look for index files and read them.
+
+        Index files can either be explicitly in the list of files to
+        ingest, or else located in the same directory as a file to ingest.
+        Index entries are always used if present.
+
+        Parameters
+        ----------
+        files : iterable over `str` or path-like objects
+            Paths to the files to be ingested.  Will be made absolute
+            if they are not already.
+
+        Returns
+        -------
+        index : `dict` [`str`, Any]
+            Merged contents of all relevant index files found. These can
+            be explicitly specified index files or ones found in the
+            directory alongside a data file to be ingested.
+        updated_files : iterable of `str`
+            Updated list of the input files with entries removed that were
+            found listed in an index file. Order is not guaranteed to
+            match the order of the files given to this routine.
+        bad_index_files: `set[str]`
+            Files that looked like index files but failed to read properly.
+        """
+        # Convert the paths to absolute for easy comparison with index content.
+        # Do not convert to real paths since we have to assume that index
+        # files are in this location and not the location which it links to.
+        files = tuple(os.path.abspath(f) for f in files)
+
+        # Index files must be named this
+        index_root_file = "_index.json"
+
+        # Group the files by directory
+        files_by_directory = defaultdict(set)
+
+        for path in files:
+            directory, file_in_dir = os.path.split(path)
+            files_by_directory[directory].add(file_in_dir)
+
+        # All the metadata read from index files with keys of full path
+        index_entries = {}
+
+        # Index files we failed to read
+        bad_index_files = set()
+
+        # Any good index files that were found and used
+        good_index_files = set()
+
+        # Look for index files in those directories
+        for directory, files_in_directory in files_by_directory.items():
+            possible_index_file = os.path.join(directory, index_root_file)
+            if os.path.exists(possible_index_file):
+                # If we are explicitly requesting an index file the
+                # messages should be different
+                index_msg = "inferred"
+                is_implied = True
+                if index_root_file in files_in_directory:
+                    index_msg = "explicit"
+                    is_implied = False
+
+                # Try to read the index file and catch and report any
+                # problems.
+                try:
+                    index = read_index(possible_index_file, force_dict=True)
+                except Exception as e:
+                    # For now only trigger the callback if the index file
+                    # was asked for explicitly.
```

Comment:
Is this "for now" a "TODO", in that it's planned to be updated later?

Reply:
I have no idea what it should do so the "for now" was "until we decided something else".  It might be that we should complain if an implicit index file is broken. Doesn't seem wrong to tell people that but also people didn't ask for it to be used anyhow.